### PR TITLE
Filtering segments by unit (only retrieve segments with the same unit…

### PR DESCRIPTION
**Pull request by gpkc**
… as the main metric).

### CODE REVIEW

File: edgar_data/XBRL/XBRL.py [hunk 1]
```diff
@@ -405,11 +405,11 @@ def _update_segments_dict(self, segments, allowed_segments_dimensions):
 
         return new_segments
 
-    def _find_segments_nodes(self, concept):
+    def _find_segments_nodes(self, concept, unit_id):
         segments = defaultdict(list)
 
         for fact_name in metrics_to_facts[concept]:
-            nodes = self._get_node_list(f"//{fact_name}")
+            nodes = self._get_node_list(f"//{fact_name}[@unitRef='{unit_id}']")
```
Comment (kut) [inline edgar_data/XBRL/XBRL.py]: is this running for all `segment_values` calls, even if there isn't duplicate? do we have performance concerns? can you profile before and after?

File: edgar_data/XBRL/Metric.py [hunk 2]
```diff
@@ -29,7 +29,8 @@ def __init__(self, value, unit_ref, xbrl, metric, calculated=False, calculated_m
         else:
             self.calculated_metric = calculated_metric
 
-        self._unit_ref = unit_ref
+        self._unit = unit
```
Comment (kut) [inline edgar_data/XBRL/Metric.py]: what if a metric doesn't have a unit? e.g. "# shares outstanding" - that's a unitless number.

File: edgar_data/XBRL/XBRL.py (hunk 1 above)
Comment (gpkc) [reply edgar_data/XBRL/XBRL.py]: Yes, I compared the timing and it doesn't make any noticeable difference. This is probably running as a somewhat optimized C code on lxml, and this is probably just one more `if` statement on the generated code, so should not impact performance really.

Comment (kut) [reply edgar_data/XBRL/XBRL.py]: ok - so this basically means that we only find segments that have the same units as the main value?

Comment (gpkc) [reply edgar_data/XBRL/XBRL.py]: Yes exactly, I think that's OK from what we discussed?

Comment (kut) [reply edgar_data/XBRL/XBRL.py]: yes

File: edgar_data/XBRL/Metric.py (hunk 2 above)
Comment (gpkc) [reply edgar_data/XBRL/Metric.py]: According to the [spec](http://www.xbrl.org/Specification/XBRL-RECOMMENDATION-2003-12-31+Corrected-Errata-2006-12-18.htm#_Toc156209151):
`All numeric items MUST have a statement of the units of measurement [...] Numeric items identify their units using the unitRef attribute.`
In the case of number of shares outstanding it is usually `xbrli:shares`.

Comment (kut) [reply edgar_data/XBRL/Metric.py]: ok

**Pull request merged**